streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java

```diff
@@ -360,6 +361,118 @@ public void shouldPutAll() {
                 rocksDBStore.get(new Bytes(stringSerializer.serialize(null, "3")))));
     }
 
+    @Test
+    public void shouldReturnKeysWithGivenPrefix() {
+        final List<KeyValue<Bytes, byte[]>> entries = new ArrayList<>();
+        entries.add(new KeyValue<>(
+                new Bytes(stringSerializer.serialize(null, "k1")),
+                stringSerializer.serialize(null, "a")));
+        entries.add(new KeyValue<>(
+                new Bytes(stringSerializer.serialize(null, "prefix_3")),
+                stringSerializer.serialize(null, "b")));
+        entries.add(new KeyValue<>(
+                new Bytes(stringSerializer.serialize(null, "k2")),
+                stringSerializer.serialize(null, "c")));
+        entries.add(new KeyValue<>(
+                new Bytes(stringSerializer.serialize(null, "prefix_2")),
+                stringSerializer.serialize(null, "d")));
+        entries.add(new KeyValue<>(
+                new Bytes(stringSerializer.serialize(null, "k3")),
+                stringSerializer.serialize(null, "e")));
+        entries.add(new KeyValue<>(
+                new Bytes(stringSerializer.serialize(null, "prefix_1")),
+                stringSerializer.serialize(null, "f")));
+
+        rocksDBStore.init((StateStoreContext) context, rocksDBStore);
+        rocksDBStore.putAll(entries);
+        rocksDBStore.flush();
+
+        final KeyValueIterator<Bytes, byte[]> keysWithPrefix = rocksDBStore.prefixScan("prefix", stringSerializer);
+        final String[] valuesWithPrefix = new String[3];
+        int numberOfKeysReturned = 0;
+
+        while (keysWithPrefix.hasNext()) {
+            final KeyValue<Bytes, byte[]> next = keysWithPrefix.next();
+            valuesWithPrefix[numberOfKeysReturned++] = new String(next.value);
+        }
+        // Since there are 3 keys prefixed with prefix, the count should be 3
+        assertEquals(3, numberOfKeysReturned);
```

Comment:
We prefer to use `assertThat()` in new test, because it is better readable. This line would then become:
 ```suggestion
        assertThat(numberOfKeysReturned, is(3));
```
Please replace also the other verifications with `assertThat()` and fix the indentation (should be 4 spaces, not 8).